### PR TITLE
Add ChibiOS HAL

### PR DESCRIPTION
**Pull request by podhrmic**
sw/airborne/arch/chibios
- no makefiles at this point (so it is structured)

### CODE REVIEW

**Comment by flixr** [comment]
Seems like the updated ChibiOS has some issues...
Do some of the added files in the HAL require this version?

**Comment by podhrmic** [comment]
Apparently some minor bug related to changes in ChibiOS/build process/board settings - probably not important right now, as the changes are not really needed. What about now?

**Comment by flixr** [comment]
You can also force push to the branch and simply remove the commit updating the submodule...

**Comment by flixr** [comment]
I think it would make sense to add the makefiles and e.g. a LisaMX board file as well, so that we can run some of the test progs using ChibiOS....

**Comment by flixr** [comment]
Source and header files should not go into conf...

File: conf/boards/lisa_mx_rt.makefile [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+# along with paparazzi; see the file COPYING.  If not, write to
+# the Free Software Foundation, 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+#
+
+# Board definition
+BOARD=lisa_mx
+BOARD_CFG=\"../../conf/boards/$(BOARD)/board.h\"#actual chibios defines
+# Arch definition
+ARCH=chibios# for backwards compatibility with the paparazzi make process
```
Comment (flixr) [inline conf/boards/lisa_mx_rt.makefile]: should be just `ARCH=chibios`

File: conf/boards/lisa_mx_rt.makefile [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+CHIBIOS_BOARD_PORT = ARMCMx/STM32F4xx/port.mk
+CHIBIOS_BOARD_LINKER = STM32F407xG.ld
+
+##############################################################################
+# Compiler settings
+#
+MCU  = cortex-m4
+
+# -----------------------------------------------------------------------
+# include Makefile.chibios instead of Makefile.stm32
```
Comment (flixr) [inline conf/boards/lisa_mx_rt.makefile]: Not needed, if `ARCH` is correctly set to `chibios`

**Comment by flixr** [comment]
Don't see where the added baro_board.h would be needed...

**Comment by flixr** [comment]
Created the rt_chibios branch for integration, merged your stuff in there (and moved the board files from conf to sw/airborne). Also ran fix_code_style.sh on the arch/chibios files...
This should make it easier to run some tests and lets anyone make pull requests against this rt_chibios branch before we merge it to master...

**Pull request closed without merging**